target/linux/ipq40xx/base-files/lib/preinit/05_set_iface_mac_ipq40xx.sh

```diff
@@ -9,6 +9,11 @@ preinit_set_mac_address() {
 		ip link set dev eth0 address $(macaddr_add "$base_mac" +1)
 		ip link set dev eth1 address $(macaddr_add "$base_mac" +3)
 		;;
+	corewav,m018u0-s1)
+		base_mac=$(mtd_get_mac_binary ART 0x1006)
+		ip link set dev eth0 address $(macaddr_add "$base_mac" +2)
+		ip link set dev eth1 address $(macaddr_add "$base_mac" +3)
```

Comment:
Have you checked whether there are matching addresses in ART 0x0 and/or 0x6? If yes, using them is preferred, if no, just keep it as it is.

Reply:
No i need keep it, because of ART 0x0 ~ 0x1000 are 0xff

Reply:
Okay.